data/src/main/kotlin/com/vultisig/wallet/data/chains/helpers/CosmosHelper.kt

```diff
@@ -113,7 +114,12 @@ class CosmosHelper(
                                 .addAllAmounts(
                                     listOf(
                                         Cosmos.Amount.newBuilder()
-                                            .setDenom(denom)
+                                            .setDenom(
+                                                if (keysignPayload.coin.contractAddress.contains("factory/") || keysignPayload.coin.contractAddress.contains("ibc/"))
+                                                    keysignPayload.coin.contractAddress
+                                                else
+                                                    denom
+                                            )
```

Comment:
https://github.com/vultisig/vultisig-ios/blob/a9c448968076cf301e91868699f10473eaf5a1ab/VultisigApp/VultisigApp/Chains/cosmos/cosmos.swift#L79

Reply:
It has been tested. Since the iOS code is slightly different, the line also includes the pointing logic.

Here’s the relevant part of the code:
https://github.com/vultisig/vultisig-ios/blob/a9c448968076cf301e91868699f10473eaf5a1ab/VultisigApp/VultisigApp/Chains/cosmos/cosmos.swift#L60-L62"

Reply:
https://finder.kujira.network/kaiyo-1/tx/39EC2EFD212EDB0C81FABAC85572254E9D706F8D99F5E4F94A090E23FA7B6EDE

https://finder.kujira.network/kaiyo-1/tx/CB54B780BEAB92F2A52A48E0C57DAC7A72245A8B29E3D173004DB8271F07E259

https://finder.kujira.network/kaiyo-1/tx/D4DB5651D85F13D7E4A09C857185D060B1209CF15F630C565C0154E3CA29C3B4